payload/builder/payload.go

```diff
@@ -173,18 +173,13 @@ func (pb *PayloadBuilder) RetrievePayload(
 	}
 
 	payload := envelope.GetExecutionPayload()
-	if !payload.IsNil() {
```

Comment:
this check does not make sense. Note that we use payload below, without checking if is nil